pkg/network/server.go

```diff
@@ -282,6 +279,48 @@ func newServerFromConstructors(config ServerConfig, chain Ledger, stSync StateSy
 	return s, nil
 }
 
+type blockAdapter struct {
+	bc StateSync
+}
+
+func (a blockAdapter) AddItem(b *block.Block) error {
+	return a.bc.AddItem(b)
```

Comment:
Statesync module itself must not implement any Item-related things. That's exactly the purpose of adapters. Use AddBlock method here.